Automattic/wp-calypso
# Refactor clipboard component to use native clipboard API

#### Changes proposed in this Pull Request

While looking at https://github.com/Automattic/wp-calypso/pull/59228, I wondered if we could remove the `clipboard` dependency. Turns out it was fairly straightforward! I also refactored the small file to Typescript along the way, as part of my goal to learn more TS. This required me to export the `ButtonProps` from `@automattic/components`. Is this the right call?

I also refactored the 2fa page to utilize this button instead of doing its own thing. 

#### Testing instructions
1. Load calypso.live
1. Find a place in calypso which uses the copy button. One example is at `/me/security/two-step`. You can click "Generate new backup codes" in the "backup codes" section.
2. Click the "copy button"
3. Make sure text was copied to your clipboard, and that other aspects of the button work (such as tooltips or `onCopy` callbacks)

<img width="351" alt="Screen Shot 2021-12-15 at 1 35 05 PM" src="https://user-images.githubusercontent.com/6265975/146271024-a3f4978e-2a0e-49d4-a947-cfa9f4cbcf0d.png">

Related to #
